### PR TITLE
Check that the target ref can be read when authorization is enabled

### LINKED ISSUE #3058: Validate read access to all reference names passed to API methods
Conceptually the Nessie authorization model relies on reference (branch/tag) names. It stands to reason that the user should have at least `READ` access to all references used in an operation.

ATM, for example, the`assignReference` method does not validate access to the `assignTo` reference, it only validates access to the reference being changed. https://github.com/projectnessie/nessie/blob/dea9dadaabe6b821814350f02ae39d2f4e4024ea/servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java#L129

This and possibly other use cases should be fixed to ensure access to all references used in API methods is validated.

### PR DESCRIPTION
**Pull request by nastra**
In this case the target ref is the one
* we assign a reference to
* merge a branch into
* transplant commits into

fixes #3058

### CODE REVIEW

**Review by snazy — DISMISSED** [review]

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java [hunk 1]
```diff
@@ -124,9 +124,12 @@ public Reference createReference(@Nullable String sourceRefName, Reference refer
...
       throws NessieNotFoundException, NessieConflictException {
-    namedRefWithHashOrThrow(assignTo.getName(), assignTo.getHash());
-
-    getAccessChecker().canAssignRefToHash(createAccessContext(), ref);
+    ServerAccessContext accessContext = createAccessContext();
+    getAccessChecker()
+        .canViewReference(
+            accessContext,
+            namedRefWithHashOrThrow(assignTo.getName(), assignTo.getHash()).getValue());
+    getAccessChecker().canAssignRefToHash(accessContext, ref);
```
Comment (ajantha-bhat) [inline servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java:132]: I think we need to handle similarly for `transplant` also. Better to check all the API once and handle it as mentioned in #3058

Comment (nastra) [reply servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java:132]: indeed, we need to check merge+transplant

File: servers/services/src/main/java/org/projectnessie/services/cel/CELUtil.java [hunk 2]
```diff
@@ -123,7 +123,7 @@
   /**
    * 'Mirrors' Nessie model objects for CEL.
    *
-   * @param model Ńessie model object
+   * @param model Nessie model object
```
Comment (dimas-b) [inline servers/services/src/main/java/org/projectnessie/services/cel/CELUtil.java:126]: :)

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java (hunk 1 above)
Comment (dimas-b) [reply servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java:132]: yes, please

Comment (nastra) [reply servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImplWithAuthorization.java:132]: fixed + added tests

**Review by ajantha-bhat — APPROVED** [review]

**Pull request merged**